### PR TITLE
Better 1.18.x support

### PR DESCRIPTION
**Pull request by JorelAli**
Adds support for 1.18.x. In particular:
- Adds support for block entities
  - Fixes chests, enderchests, bells, conduits not rendering
- Adds support for inventory saving for containers:
  - Chests, double chests and barrels
  - Dispensers and Droppers
  - Furnaces (including blast furnace and smoker)
  - Hoppers
  - Lecterns (limited support)
- Adds support for other entities:
  - Cats now store owners and their type
  - Villagers now store their profession, type and whether they have NoAI set
  
Other notes:
- Tested on 1.18.2 (and used to backup a 1.18.2 Minecraft server on the brink of destruction, so thanks for creating this tool!)

### CODE REVIEW

**Comment by mircokroon** [comment]
Thank you! This looks like quite a bit of stuff so I'll go over it more when I have some time.

**Comment by JorelAli** [comment]
I forgot to mention the following:
- I couldn't get beds to work. Beds will appear in the saved world, but won't be rendered. Bed types are stored (because it's all part of their block state), but a block update is required to make them visible again.
- Signs work
- Lecterns _sometimes_ work. Since they classify as containers, you have to interact with them to save them. When testing, I was able to save some books, but wasn't able to save other books. The `/data get block ...` command shows that the book NBT is stored in the downloaded world, but I probably wrote it in the wrong place(?)

Other things I found:
- I tried getting item frame rotation to work, but to no avail. (wiki.vg says it's stored in index 9, but your code uses index 7). I didn't investigate this that much (item frame rotations weren't important to me), so I left it as it was.

Known bugs (not sure if this is introduced from my changes, or were bugs that have always been present. Either way, these are some things that I found and don't think they're worth making a million GitHub issues, so I'll just list then there. Feel free to either make GitHub issues or ask me to make GitHub issues if you think it's worth it and I'll attempt to get some more reproducible results for the following bugs):
- Sometimes, disconnecting at a location will corrupt the chunk you disconnected in. This typically makes the entire chunk empty. Loading the saved world on a paper server makes paper complain that the chunk is invalid. I'm not sure if this is because the world downloader is mid-saving a chunk while I'm connected? I've always done "leave the server, then right click the GUI and save and exit". It's possible that doing this the other way round fixes this issue
- For some reason, armor stands and item frames are missing metadata. Armor stand invisibility flags, pose and whether their custom name is visible doesn't save. Item frame visibility doesn't save. I don't know why.
- (Possibly a bug, I never formally tested this one) Loading the world downloader on a world twice (e.g. log into a server, run around, quit, log in again, run around, quit) will mess up entities (I believe?) by storing their data twice. Loading the saved world on a paper server complains about duplicate entity UUIDs

**Comment by JorelAli** [comment]
This PR should address #327, #339, #313, #305

Also works on #298, should fix #286, should fix the issue by @Alacaster in #288 and improves on #234.

I happen to have thoroughly enjoyed updating this tool (albeit having deal with the stress of trying to do it quickly due to a dying Minecraft server) and am willing to address some of the many issues raised here on GitHub (e.g. villager trades, command block NBT, etc). I wanted to add support for paintings, but didn't have enough time to do so. If you'd like to collaborate on this tool, I'm more than happy to do so! (If not, I'm also happy to send PRs if and when I get round to implementing certain features)

**Comment by JorelAli** [comment]
@Ravendwyr Good spot - my bad. Once again, the use of copy-paste has screwed me over again! Should be fixed now.

**Comment by Ravendwyr** [comment]
Aye, everything seems ok at first glance. Sorry for the trouble. 😅

**Comment by JorelAli** [comment]
> I see, may I ask how long this "some time" may be? As I had some containers closed roughly 5 to 10 minutes before "saving & exit" and opening the world up.

@AgileGhost Approximately 5 seconds, no longer than 15 seconds. Chunk saving runs on a timer, I'm not sure where or what the timer duration is.

> Anyways whatever is causing it, best of luck, and hope it's an easy fix.

When interacting with a container, if you've enabled info messages (Right click the UI > Settings > General > Send info messages), you should get a popup in the action bar in-game about whether the container was saved or not (either it was recorded, or something went wrong). If it says it's been recorded and it wasn't, then that's something very broken. If it's red saying something went wrong, hopefully that'll be easy to fix 🤞

> This is the error I got when I was interacting with the hoppers and placing down the dropper.

I'll make sure to keep an eye out for that error. I'll also make sure to poke lots of hoppers and place lots of droppers 😅

**Comment by Ravendwyr** [comment]
> I see, may I ask how long this "some time" may be? As I had some containers closed roughly 5 to 10 minutes before "saving & exit" and opening the world up. I believe this should have easily been long enough for it to save, but I could be wrong. I did also test chests with items already in it and chests where I messed with the items, but none worked.

Initially I had a few red messages saying that the chunk wasn't loaded when attempting to save chest inventories. Going elsewhere on the map (like, a good number of chunks away) then going back again solved that though. Would this help?

**Comment by JorelAli** [comment]
@AgileGhost @Ravendwyr Let's continue this over at #355, so we're not cluttering this PR. _(Sorry for the hassle @mircokroon!)_

**Comment by JorelAli** [comment]
Updated so far:
- Fixes `ArrayIndexOutOfBoundsException: arraycopy: last source index 10 out of bounds for byte[2]` bug (see [here](https://github.com/mircokroon/minecraft-world-downloader/pull/354#issuecomment-1142066927))
- Fixes beds not rendering properly
- Fixes the following:
  - #316
- _Should_ fix the following (but it may not):
  - #330 
  - #297 
- Implements the following:
  - #245

File: src/main/java/game/data/chunk/version/Chunk_1_18.java [hunk 1]
```diff
@@ -69,66 +62,70 @@ protected void parse(DataTypeProvider dataProvider) {
...
-            setChunkSection(sectionY, section);
-
-            // For some reason that I can't figure out, servers don't include containers
-            // in the list of block_entities. We need to know that these block entities
-            // exist, otherwise we'll end up not writing block entity data which results
-            // in certain block entities that don't render!
-            for(SpecificTag tag : blockPalette.toNbt()) {
-                // Only iterate over the chunk section if the palette contains a block entity
-                String blockName = tag.get("Name").stringValue();
-                if(isBlockEntity(blockName)) {
```
Comment (JorelAli) [inline src/main/java/game/data/chunk/version/Chunk_1_18.java]: The reason I had this additional iteration and if condition here is that I wanted to only iterate over the chunk section if there was a valid block entity in that chunk section - this prevents unnecessarily iterating over every block in the chunk section if there isn't one (e.g. don't iterate over a chunk section which consists of only air)

File: src/main/java/game/data/entity/specific/Item.java [hunk 2]
```diff
@@ -26,6 +26,7 @@ public Item() {
     protected void addNbtData(CompoundTag root) {
         super.addNbtData(root);
 
+        // TODO: make option in menu for whether items should despawn or stay permanently
```
Comment (JorelAli) [inline src/main/java/game/data/entity/specific/Item.java:29]: Great idea ❤️

**Comment by JorelAli** [comment]
@mircokroon Thanks for the update! Moving the registries out of the world manager is a much desired change, thanks.

> I think block placements are still broken for 1.18 right? At least the SectionBlocksUpdate one, I got a bunch of errors with that enabled so I added the _ to the packet ID file to disable that until someone gets around to fixing it.

Honestly, I had no idea that this was broken - could you give an example of what I'd need to do to reproduce the errors you got? I'll have a look into this for you if you'd like.

> I guess we should also update this page with command blocks and the new entities

Sounds good to me. As briefly mentioned in #355, I'll leave villager trades for another PR or something - it's a lot of work and I don't want to risk making anything unstable at this point.

> It's a bit funny how the only pet we now properly support are cats, maybe a dog lover will come around and implement those at some point.

The main reason I added support for cats was because they have type metadata, and when trying to back up my Minecraft server, I didn't want all of my cats turning into tabby cats! I'm more than happy to add support for axolotls (which also have types), horses (horse types) and sheep (sheep color), since I believe they should take priority. I could also throw in dogs and stare at other entities, but I'll get round to it when I get round to it.

**Comment by mircokroon** [comment]
> Honestly, I had no idea that this was broken - could you give an example of what I'd need to do to reproduce the errors you got? I'll have a look into this for you if you'd like.

Hmm I actually can't reproduce the error on a vanilla server now, seems to work fine there. It triggers immediately when joining Hypixel, not sure if they're doing something weird or if it's an edge case of some kind. 

```
Exception in thread "Chunk Parser Service" java.lang.ArrayIndexOutOfBoundsException: Index 271 out of bounds for length 256
	at game.data.chunk.version.encoder.BlockLocationEncoder_1_16.write(BlockLocationEncoder_1_16.java:29)
	at game.data.chunk.version.ChunkSection_1_18.setBlockAt(ChunkSection_1_18.java:92)
	at game.data.chunk.Chunk.updateBlock(Chunk.java:491)
	at game.data.chunk.version.Chunk_1_16.updateBlocks(Chunk_1_16.java:71)
	at game.data.WorldManager.lambda$multiBlockChange$6(WorldManager.java:686)
```

**Comment by JorelAli** [comment]
Lecterns are now fixed.

**Comment by JorelAli** [comment]
Wait, I want axolotls, horses and sheep first.

**Comment by JorelAli** [comment]
Adds support for horses, axolotls and sheep. Doesn't add support for horse armor or custom saddles (but if the horse is saddled, it'll save a default saddle on the horse).

That finalizes everything I wanted to add in this PR (yes, I promise I won't shove more features into this! Sorry for the feature creep!).

File: src/main/java/game/data/entity/specific/Villager.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import se.llbit.nbt.StringTag;
+import se.llbit.nbt.Tag;
+
+/**
+ * Handle villagers because they have interesting metadata.
+ */
+public class Villager extends MobEntity {
+    private VillagerMetaData metaData;
+    private List<VillagerTrade> trades;
+    private int villagerLevel;
```
Comment (mircokroon) [inline src/main/java/game/data/entity/specific/Villager.java:27]: I think this is unused currently, should this be written to NBT as well?

Comment (JorelAli) [reply src/main/java/game/data/entity/specific/Villager.java:27]: The reason I didn't use the `villagerLevel` from the trade packet is because there appears to also be another villager level (`VillagerMetadata:122`) stored in `VillagerData.level` which is populated when the entity is added to the chunk.

I'm not entirely sure if these are both the same villager level and/or whether this needs updating. If so, then probably?

**Comment by JustASpeedrunner** [comment]
Getting errors trying to compile


Full output of mvn package:
```
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for mircokroon.minecraft-world-downloader:minecraft-world-downloader:jar:1.0.0.0
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.openjfx:javafx-controls:jar -> version 15.0.1 vs 11 @ line 261, column 21
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] 
[INFO] --< mircokroon.minecraft-world-downloader:minecraft-world-downloader >--
[INFO] Building minecraft-world-downloader 1.0.0.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ minecraft-world-downloader ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 21 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ minecraft-world-downloader ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ minecraft-world-downloader ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 6 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ minecraft-world-downloader ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M1:test (default-test) @ minecraft-world-downloader ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running game.data.chunk.ChunkTest
[ERROR] Tests run: 25, Failures: 0, Errors: 7, Skipped: 0, Time elapsed: 0.132 s <<< FAILURE! - in game.data.chunk.ChunkTest
[ERROR] chunk_1_12  Time elapsed: 0.038 s  <<< ERROR!
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.IllegalStateException: Internal problem occurred, please report it. Mockito is unable to load the default implementation of class that is a part of Mockito distribution. Failed to load interface org.mockito.plugins.MockMaker
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.reflect.InvocationTargetException
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.IllegalStateException: Failed to resolve the class file version of the current VM: This JVM's version string does not seem to be valid: 18-ea
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)

[ERROR] chunk_1_13  Time elapsed: 0 s  <<< ERROR!
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_13(ChunkTest.java:103)
Caused by: java.lang.IllegalStateException: Internal problem occurred, please report it. Mockito is unable to load the default implementation of class that is a part of Mockito distribution. Failed to load interface org.mockito.plugins.MockMaker
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.reflect.InvocationTargetException
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.IllegalStateException: Failed to resolve the class file version of the current VM: This JVM's version string does not seem to be valid: 18-ea
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)

[ERROR] chunk_1_14  Time elapsed: 0 s  <<< ERROR!
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_14(ChunkTest.java:108)
Caused by: java.lang.IllegalStateException: Internal problem occurred, please report it. Mockito is unable to load the default implementation of class that is a part of Mockito distribution. Failed to load interface org.mockito.plugins.MockMaker
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.reflect.InvocationTargetException
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.IllegalStateException: Failed to resolve the class file version of the current VM: This JVM's version string does not seem to be valid: 18-ea
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)

[ERROR] chunk_1_15  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_15(ChunkTest.java:113)
Caused by: java.lang.IllegalStateException: Internal problem occurred, please report it. Mockito is unable to load the default implementation of class that is a part of Mockito distribution. Failed to load interface org.mockito.plugins.MockMaker
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.reflect.InvocationTargetException
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.IllegalStateException: Failed to resolve the class file version of the current VM: This JVM's version string does not seem to be valid: 18-ea
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)

[ERROR] chunk_1_16  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_16(ChunkTest.java:118)
Caused by: java.lang.IllegalStateException: Internal problem occurred, please report it. Mockito is unable to load the default implementation of class that is a part of Mockito distribution. Failed to load interface org.mockito.plugins.MockMaker
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.reflect.InvocationTargetException
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.IllegalStateException: Failed to resolve the class file version of the current VM: This JVM's version string does not seem to be valid: 18-ea
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)

[ERROR] chunk_1_17  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)
	at game.data.chunk.ChunkTest.testForWithLight(ChunkTest.java:65)
	at game.data.chunk.ChunkTest.chunk_1_17(ChunkTest.java:158)
Caused by: java.lang.IllegalStateException: Internal problem occurred, please report it. Mockito is unable to load the default implementation of class that is a part of Mockito distribution. Failed to load interface org.mockito.plugins.MockMaker
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.reflect.InvocationTargetException
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.IllegalStateException: Failed to resolve the class file version of the current VM: This JVM's version string does not seem to be valid: 18-ea
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)

[ERROR] chunk_1_16_light  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_16_light(ChunkTest.java:123)
Caused by: java.lang.IllegalStateException: Internal problem occurred, please report it. Mockito is unable to load the default implementation of class that is a part of Mockito distribution. Failed to load interface org.mockito.plugins.MockMaker
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.reflect.InvocationTargetException
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)
Caused by: java.lang.IllegalStateException: Failed to resolve the class file version of the current VM: This JVM's version string does not seem to be valid: 18-ea
	at game.data.chunk.ChunkTest.testFor(ChunkTest.java:34)
	at game.data.chunk.ChunkTest.chunk_1_12(ChunkTest.java:98)

[INFO] Running packets.builder.PacketBuilderAndParserTest
[INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in packets.builder.PacketBuilderAndParserTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   ChunkTest.chunk_1_12:98->testFor:34 » IllegalState Could not initialize plugin...
[ERROR]   ChunkTest.chunk_1_13:103->testFor:34 » IllegalState Could not initialize plugi...
[ERROR]   ChunkTest.chunk_1_14:108->testFor:34 » IllegalState Could not initialize plugi...
[ERROR]   ChunkTest.chunk_1_15:113->testFor:34 » IllegalState Could not initialize plugi...
[ERROR]   ChunkTest.chunk_1_16:118->testFor:34 » IllegalState Could not initialize plugi...
[ERROR]   ChunkTest.chunk_1_16_light:123->testFor:34 » IllegalState Could not initialize...
[ERROR]   ChunkTest.chunk_1_17:158->testForWithLight:65 » IllegalState Could not initial...
[INFO] 
[ERROR] Tests run: 43, Failures: 0, Errors: 7, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.680 s
[INFO] Finished at: 2022-06-02T10:05:25-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M1:test (default-test) on project minecraft-world-downloader: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/jas/Desktop/minecraft-world-downloader/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

File: src/main/java/game/data/entity/specific/Villager.java (hunk 3 above)
Comment (JorelAli) [reply src/main/java/game/data/entity/specific/Villager.java:27]: As far as I'm aware, the villager's xp is a sufficient representation of the villager's "level" and it's almost as if the villager's level isn't even needed.

**Comment by JorelAli** [comment]
> Getting errors trying to compile

@JustASpeedrunner Sounds like a malformed Maven dependency - I can't reproduce this on my machine and am able to compile everything normally (and also, GitHub actions is able to build it correctly, so it's probably something dodgy on your end, not ours). Are you sure that Maven has resolved (and downloaded) the dependencies correctly? Consider purging any local dependencies before rebuilding:

```sh
mvn dependency:purge-local-repository
mvn clean install
```

**Comment by JustASpeedrunner** [comment]
Odd, I tried those commands, the first of which worked and the second of which failed. Also just to make sure attempted reinstalling maven which still did not help, not sure what the cause is but I'll go look for some maven focused forum answers instead of asking here to make sure I don't clog it up here. Thanks!

**Comment by mircokroon** [comment]
Closes #251
Closes #279 
Closes #280 
Closes #286 
Closes #288 
Closes #297 
Closes #298 
Closes #305 
Closes #311
Closes #313 
Closes #315
Closes #316
Closes #327 
Closes #330 
Closes #332 
Closes #339 
Closes #345

**Comment by mircokroon** [comment]
Closes #251
Closes #279 
Closes #280 
Closes #286 
Closes #288 
Closes #297 
Closes #298 
Closes #305 
Closes #311
Closes #313 
Closes #315
Closes #316
Closes #327 
Closes #330 
Closes #332 
Closes #339 
Closes #345

**Comment by mircokroon** [comment]
That apparently doesn't work, ok then

**Comment by JorelAli** [comment]
@mircokroon This should also close #245

**Comment by mircokroon** [comment]
Oh yeah, looks like I missed some
